### DISCUSSION TITLE
How to share a screen between tabs while keeping the tab menu on all screens

### DISCUSSION THREAD

**Discussion by andrewdang17** [react-navigation/react-navigation#10456] [Q&A] [answered]
Say I have 3 tabs (home, search and library), and within each of those tabs I can access a Book screen. My first thought was to create the Tab.Navigator separately and pass that to my main navigation stack like so:

```
const TabNav = () => (
    <Tab.Navigator>
      <Tab.Screen name="home" component={Home} />
      <Tab.Screen name="search" component={Search} />
      <Tab.Screen name="library" component={Library} />
    </Tab.Navigator>
)

const AppStack = () => (
   <Stack.Navigator>
     <Stack.Screen name="main" component={TabNav} />
     <Stack.Screen name="book" component={Book} />
   </Stack.Navigator>
)
```

However, if I visit the book screen from any of the tab components the tab menu does not appear. I've tried reading on nested navigation from the docs but all the examples assume each Tab has its own set of unique screens.

**Comment by andrewdang17** [comment]
If I create Navigators for each Tab.Screen and duplicate the same screens, it results in ` WARN  Found screens with the same name nested inside one another.` so it seems like sharing is not possible.

**Comment by raajnadar** [comment, marked as answer]
You need to do the reverse nest the stack inside the tab navigator.

```
const HomeStack = () => (
   <Stack.Navigator>
     <Stack.Screen name="main" component={TabNav} />
     <Stack.Screen name="HomeBook" component={Book} />            <------- check this
   </Stack.Navigator>
)

// same as above for the other to tabs
// Use the Book component as same but name different (Same name should work)

const TabNav = () => (
    <Tab.Navigator>
      <Tab.Screen name="home" component={HomeStack} />
      <Tab.Screen name="search" component={SearchStack} />
      <Tab.Screen name="library" component={LibraryStack} />
    </Tab.Navigator>
)
```

**Reply by andrewdang17** [reply]
@raajnadar that works, thank you!
